app/javascript/components/users/SignInPage.tsx

```diff
@@ -80,152 +78,149 @@ function SignInPage(_props:SignInPageProps) : JSX.Element {
         backgroundColor: "#3f51b5",
       },
       appbar: {
-        background: grey[200],
+        backgroundColor: theme.palette.action.hover, 
       },
       responsive: {
         [theme.breakpoints.down('sm')]: {
           width: "100%",
           marginTop: 45,
           marginBottom: 45
-          },
-          [theme.breakpoints.up('lg')]: {
-            margin: 75,
-          },
+        },
+        [theme.breakpoints.up('lg')]: {
+          margin: 75,
+        },
+        [theme.breakpoints.down('md')]: {
+          marginTop: 45,
+          marginBottom: 45
+        },
       },
       paper: {
         margin: `${theme.spacing(1)}px auto`,
         padding: theme.spacing(2),
         borderRadius: 15,
+        minHeight: 500,
       },
-		}),
-		);
+    }),
+  );
 
-  //Error boundary
-  class ErrorBoundary extends React.Component {
-    constructor(props) {
-      super(props);
-    }
-    state = {
-      errorMessage: 'Something went wrong. Reload the page'
-    }
-    static getDerivedStateFromError(error) {
-      // Update state so the next render will show the fallback UI.
-      return { hasError: true };
-    }
-    componentDidCatch(error, errorInfo) {
-      // You can also log the error to an error reporting service
-      this.logErrorToServices(error, errorInfo);
-    }
-    logErrorToServices = console.log
-    render() {
-      if (this.state.errorMessage) {
-        return (
-          <p>
-            {this.state.errorMessage}
-          </p>
-        )
-      }
-      return this.props.children; 
-    }
-  }
-
-	//Setting up error messages
+  const termsPath = routes.static_terms_and_privacy_path;
+  
+  
+  //Setting up error messages
   const classes = useStyles();
   const { formatMessage } = useIntl();
-	const yup = useYup();
-  const loginHeaderLabel = formatMessage({id: 'login.header'});
-  const forgotPasswordlabel = formatMessage({id: 'login.forgot_password'});
-  const copyright = formatMessage({id: 'footer.copyright'});
-  const terms = formatMessage({id: 'footer.terms_and_privacy'});
-  const getStartedLabel = formatMessage({id: 'login.get_started'});
+  const yup = useYup();
+  const loginHeaderLabel = formatMessage({ id: 'login.message' });
+  const forgotPasswordlabel = formatMessage({ id: 'login.forgot_password' });
+  const copyright = formatMessage({ id: 'footer.copyright' });
+  const terms = formatMessage({ id: 'footer.terms_and_privacy' });
+  const getStartedLabel = formatMessage({ id: 'login.get_started' });
+  const errorBoundaryLabel = formatMessage({ id: 'login.errors.errorBoundary' });
 
-	return (
-		<Grid container spacing={0}>
-			<Grid item xs={12}>
-      <div className={classes.root}>
+  //Error Boundary
+  function Fallback() {
+    return <div>{errorBoundaryLabel}</div>
+  }
```

Comment:
I've never seen anyone define a component inside another component like this. Does this work?

Reply:
It works. Should we modify it?

Reply:
I think it would probably be best to have the Fallback component defined in the same file for now but not defined inside the SignInPage function.  There are two main reasons:

* We may need to move Fallback to its own file at some point since it may become a more complex component.
* I think there defining it in here might defeat some of the optimizations used by React